src/ScriptBuilder/Saga/Writers/OracleSagaScriptWriter.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using NServiceBus.Persistence.Sql.ScriptBuilder;
+
+class OracleSagaScriptWriter : ISagaScriptWriter
+{
+    TextWriter writer;
+    SagaDefinition saga;
+    string tableName;
+
+    public OracleSagaScriptWriter(TextWriter textWriter, SagaDefinition saga)
+    {
+        writer = textWriter;
+        this.saga = saga;
+        if (saga.TableSuffix.Length > 27)
+        {
+            throw new Exception($"Saga '{saga.TableSuffix}' contains more than 27 characters, which is not supported by SQL persistence using Oracle. Either disable Oracle script generation using the SqlPersistenceSettings assembly attribute, shorten the name of the saga, or specify an alternate table name using the SqlSagaAttribute's 'tablePrefix' parameter.");
+        }
+        if (Encoding.UTF8.GetBytes(saga.TableSuffix).Length != saga.TableSuffix.Length)
```

Comment:
I think checking each `char` of the string if it is higher than 255 (or 127, not sure which set Oracle supports) would be cleaner and would be equivalent.